draft-ietf-quic-recovery.md

```diff
@@ -398,8 +417,82 @@ A packet sent on an RTO alarm MUST NOT be blocked by the sender's congestion
 controller. A sender MUST however count these bytes as additional bytes in
 flight, since this packet adds network load without establishing packet loss.
 
+## Multiple Packet Number Space Optimizations {#optimizations}
+
+There are cases where one may be able to gain recovery information from
+acknowledgements of packets in another packet number space, but they rely
+on complex assumptions about the peer’s processing and acknowledgement
+algorithms.  Even those are unable to quickly recover from cases such as
+losing the client's Initial, but receiving the 0-RTT packets.  Below are
+three different optimizations in increasing complexity that minimize
+handshake latency.
+
+### EMPTY_ACK frames
+
+The EMPTY_ACK frame indicates a packet for the connection was received,
+but it could not be processed, because the decryption keys are not yet
+available. EMPTY_ACK enables faster recovery of lost Initial
+and Handshake packets when the only other outstanding packets are
+undecryptable.
+
+The receiver SHOULD send an EMPTY_ACK frame soon (e.g., 1ms) after
+undecryptable packets are received, even if those received packets are
+not buffered for later decryption.  The small delay allows for cases when
+0-RTT packets are reordered in front of the Initial, which is not uncommon
+on networks that prioritize small packets.  The receiver should limit the
+number of EMPTY_ACK frames sent to one per packet number space per RTT.
+If no RTT is known, only one per encryption level should be sent.
+
+When an EMPTY_ACK frame is received, a sender should immediately
+retransmit the missing handshake packet(s) as though the handshake timer
+fired and re-arm the handshake timer when the handshake packets are sent.
+If the missing handshake data was timer retransmitted after the packets
+that triggered the empty ack, then the empty ack should be ignored.
+
+An EMPTY_ACK frame does not acknowledge a new packet, and in cases
+when multiple packets are outstanding, the RTT signal is ambiguous,
+so it should not be used like an RTT signal from a newly acknowledged
+packet.  It MAY change the connection’s default RTT if no RTT measurements
+have been taken.
+
```

Comment:
OK. What happens when some smart ass hacker sends bogus packets that "fail to decrypt"? Does it automatically force the receiver to send reply packets with empty acks?